napari/_qt/layer_controls/widgets/qt_colormap_control.py

```diff
@@ -0,0 +1,198 @@
+            style.rect.width() - TEXT_WIDTH,
+            style.rect.y() + PADDING,
+            style.rect.width(),
+            style.rect.height() - 2 * PADDING,
+        )
+        style2.rect = text_rect
+        super().paint(painter, style2, model)
+        name = display_name_to_name(model.data())
+        cbar = make_colorbar(ensure_colormap(name), (18, 100))
+        image = QImage(
+            cbar,
+            cbar.shape[1],
+            cbar.shape[0],
+            QImage.Format_RGBA8888,
+        )
+        painter.drawImage(cbar_rect, image)
+
+    def sizeHint(self, style: QStyleOptionViewItem, model: QModelIndex):
+        res = super().sizeHint(style, model)
+        res.setHeight(self.base_height)
+        res.setWidth(max(500, res.width()))
+        return res
+
+
+class QtColormapComboBox(QComboBox):
+    """Combobox showing colormaps
+
+    Parameters
+    ----------
+    parent : QWidget
+        Parent widget of comboxbox.
+    """
+
+    def __init__(self, parent) -> None:
+        super().__init__(parent)
+        view = QListView()
+        view.setMinimumWidth(COLORMAP_WIDTH + TEXT_WIDTH)
+        view.setItemDelegate(ColorStyledDelegate(ENTRY_HEIGHT))
+        self.setView(view)
+
+
+class QtColormapControl(QtWidgetControlsBase):
+    """
+    Class that wraps the connection of events/signals between the layer colormaps
+    attribute and Qt widgets.
+
+    Parameters
+    ----------
+    parent: qtpy.QtWidgets.QWidget
+        An instance of QWidget that will be used as widgets parent
+    layer : napari.layers.Layer
+        An instance of a napari layer.
+
+    Attributes
+    ----------
+    colorbarLabel : qtpy.QtWidgets.QLabel
+        Label text of colorbar widget.
+    colormapComboBox : qtpy.QtWidgets.QComboBox
+        ComboBox controlling current colormap of the layer.
+    colormapWidget : qtpy.QtWidgets.QWidget
+        Widget to wrap combobox and label widgets related with the layer colormap attribute.
+    colormapWidgetLabel : napari._qt.layer_controls.widgets.qt_widget_controls_base.QtWrappedLabel
+        Label for the color mode chooser widget.
+    """
+
+    def __init__(self, parent: QWidget, layer: Layer) -> None:
+        super().__init__(parent, layer)
+        # Setup layer
+        self._layer.events.colormap.connect(self._on_colormap_change)
+
+        # Setup widgets
+        comboBox = QtColormapComboBox(parent)
+        comboBox.setObjectName('colormapComboBox')
+        comboBox._allitems = set(self._layer.colormaps)
+
+        for name, cm in AVAILABLE_COLORMAPS.items():
+            if name in self._layer.colormaps:
+                comboBox.addItem(cm._display_name, name)
+
+        comboBox.currentTextChanged.connect(self.changeColor)
+        self.colormapComboBox = comboBox
+        self.colorbarLabel = QLabel(parent=parent)
+        self.colorbarLabel.setObjectName('colorbar')
+        self.colorbarLabel.setToolTip(trans._('Colorbar'))
+
+        colormap_layout = QHBoxLayout()
```

Comment:
```suggestion
        colormap_layout = QHBoxLayout()
        colormap_layout.setContentsMargins(0, 0, 0, 2)
```
Fixes layout issue I noticed.